src/software/thunderscope/replay/proto_logger.py

```diff
@@ -67,7 +72,7 @@ def __init__(self, log_path, log_prefix="protolog_", time_provider=None):
         self.start_time = self.time_provider()
         self.stop_logging = False
 
-    def __enter__(self):
+    def __enter__(self) -> "ProtoLogger":
```

Comment:
I think you could return `Self` (capital S) here instead of a string: https://realpython.com/python-type-self/
```suggestion
    def __enter__(self) -> Self:
```

Reply:
A better way would be to import `from __future__ import annotations` and then you can use the actual type
```suggestion
    def __enter__(self) -> ProtoLogger:
```

Reply:
We're not on python 3.11 yet so I think we can't do this. Alternative would be to `from __future__ import annotations` and then you can use the actual type

Reply:
removed